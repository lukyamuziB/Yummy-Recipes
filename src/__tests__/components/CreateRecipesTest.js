import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ReactDom from 'react-dom';
import configureMockStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-dom';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import CreateRecipe from '../../components/createRecipes2';

Enzyme.configure({ adapter: new Adapter() });

describe('<CreateRecipe />', () => {
  const store = configureMockStore([thunk])({
    categories: {},
  });
  const props = {
    createRecipes: jest.fn(() => Promise.resolve('fetchRecipes')),
    history: { push: jest.fn() },
    match: {
      param: {
        id: 1,
      },
    },
  };
  it('should render without crashing', () => {
    const { enzymeWrapper } = mount(
      <Provider store={store}>
        <MemoryRouter>
          <CreateRecipe {...props} />
        </MemoryRouter>
      </Provider>,
    );
  });
  it('renders fully', () => {
    const tree = renderer.create(
      mount(
        <Provider store={store}>
          <MemoryRouter>
            <CreateRecipe {...props} />
          </MemoryRouter>
        </Provider>,
      ),
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
