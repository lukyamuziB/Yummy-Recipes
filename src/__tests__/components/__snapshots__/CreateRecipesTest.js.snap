// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateRecipe /> renders fully 1`] = `
<div>
  <div>
    <div
      className="modal modal-fixed-footer"
      id="modal4undefined"
    >
      <div
        className="modal-content"
      >
        <div
          className="conatiner"
        >
          <h5>
             Add Recipe to 
             Category
          </h5>
          <form
            onSubmit={[Function]}
          >
            <div
              className="form-group %{touched && error ? 'has-danger':''}"
            >
              <lable>
                Name
              </lable>
              <input
                className="form-control"
                name="name"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                type="text"
                value=""
              />
              
            </div>
            <div
              className="form-group %{touched && error ? 'has-danger':''}"
            >
              <lable>
                Description
              </lable>
              <input
                className="form-control"
                name="description"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                type="text"
                value=""
              />
              
            </div>
            <button
              type="submit"
            >
               Create Recipe 
            </button>
          </form>
          <div
            className="modal-footer"
          >
            <p
              className="modal-action modal-close waves-effect waves-green btn-flat "
            >
              Cancel
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
